### PR TITLE
[3.10] imagebuilder fails for certain COPY statements

### PR DESCRIPTION
**Pull request by smarterclayton**
openshift-ansible can't build via imagebuilder due to `COPY dir /` copying to `/dir/...` instead of `/...`

Ref: openshift/imagebuilder#79

### CODE REVIEW

**Comment by smarterclayton** [comment]
@bparees @liggitt without this certain images won't build correctly on OpenShift.

File: glide.lock [hunk 1]
```diff
@@ -393,7 +393,7 @@ imports:
   subpackages:
   - log
 - name: github.com/emicklei/go-restful-swagger12
-  version: 5e28dc7157ebd9247a268c26dd1bee7818aa33f7
+  version: 76d80a40857969dddc1a4c21d1eb82f7ea20e9f5
```
Comment (liggitt) [inline glide.lock]: I don't know where this sha came from, but it's not in our repo: https://github.com/openshift/emicklei-go-restful-swagger12/commit/76d80a40857969dddc1a4c21d1eb82f7ea20e9f5

Comment (liggitt) [reply glide.lock]: the old sha was correct

File: glide.lock [hunk 2]
```diff
@@ -1073,7 +1073,7 @@ imports:
 - name: github.com/sirupsen/logrus
   version: 89742aefa4b206dcf400792f3bd35b542998eb3b
 - name: github.com/skynetservices/skydns
-  version: 775ef406696380ce92b76e72421e86b6868905d8
+  version: d3e54b74891269f7d7aa1814af2ca080b294dd29
```
Comment (liggitt) [inline glide.lock]: same here - https://github.com/openshift/skynetservices-skydns/commit/d3e54b74891269f7d7aa1814af2ca080b294dd29

File: vendor/k8s.io/kubernetes/pkg/kubectl/cmd/cp.go [hunk 3]
```diff
@@ -244,6 +244,7 @@ func copyFromPod(f cmdutil.Factory, cmd *cobra.Command, cmderr io.Writer, src, d
 // stripPathShortcuts removes any leading or trailing "../" from a given path
 func stripPathShortcuts(p string) string {
 	newPath := path.Clean(p)
+	newPath = strings.TrimLeft(p, "../")
```
Comment (liggitt) [inline vendor/k8s.io/kubernetes/pkg/kubectl/cmd/cp.go]: pretty sure this is reverting https://github.com/openshift/origin/pull/20034, though I have questions about that pick

**Comment by smarterclayton** [comment]
GCP jenkins job your cranky days are numbered

File: glide.lock (hunk 1 above)
Comment (smarterclayton) [reply glide.lock]: I'll clear my glide cache.  I just ran restore.

Comment (smarterclayton) [reply glide.lock]: Yeah, that dropped it from glide.

File: vendor/k8s.io/kubernetes/pkg/kubectl/cmd/cp.go (hunk 3 above)
Comment (smarterclayton) [reply vendor/k8s.io/kubernetes/pkg/kubectl/cmd/cp.go]: That pick is in the origin-3.10-kubernetes-1.10.2 branch, but not picked into origin 3.10 vendor, so it either should be in this branch, or not in the origin-3.10-kubernetes-1.10.2 branch.

Comment (liggitt) [reply vendor/k8s.io/kubernetes/pkg/kubectl/cmd/cp.go]: Ah, it's picking the fix up from master.

**Comment by smarterclayton** [comment]
/retest 

may be a real problem, but i don't know how

**Comment by smarterclayton** [comment]
Need to clear the job status...

**Comment by liggitt** [comment]
/hold
getting the kube branches straightened out so release-3.10 is the source for origin-3.10-kubernetes-1.10.2
will unhold momentarily

**Comment by smarterclayton** [comment]
The image_ecosystem failure seems to be 3.10 branch associated - https://github.com/openshift/origin/pull/20069 is failing too.

**Comment by smarterclayton** [comment]
Rebuilding it.

**Comment by smarterclayton** [comment]
labelling

**Pull request merged**